### PR TITLE
Fix bug on mysql passwords

### LINKED ISSUE #137: Mysql error on installation using defaults
Error discovered using molecule tests.

```
   TASK [aalaesar.nextcloud : [mySQL] - Update mysql root password] ***************
  fatal: [instance]: FAILED! => {"changed": false, "msg": "unable to connect to database, check login_user and login_password are correct or /etc/mysql/debian.cnf has the credentials. Exception message: (2003, \"Can't connect to MySQL server on 'localhost' ([Errno 99] Cannot assign requested address)\")"}
```

### PR DESCRIPTION
**Pull request by staticdev**
Depends on molecule PR https://github.com/aalaesar/install_nextcloud/pull/136

Closes #137

### CODE REVIEW

File: tasks/setup_env.yml [hunk 1]
```diff
@@ -57,3 +57,8 @@
   when:
     - nc_sudo_installed_result.rc is defined
     - nc_sudo_installed_result.rc != 0
+
+- name: "[ENV] - Define mysql_daemon."
+  set_fact:
```
Comment (wiktor2200) [inline tasks/setup_env.yml:62]:
```suggestion
  ansible.builtin.set_fact:
```

File: tasks/db_mysql.yml [hunk 2]
```diff
@@ -12,7 +12,11 @@
   vars:
     nc_mysql_deps:
       - "php{{ php_ver }}-mysql"
-      - "python{{ '3' if ansible_python.version.major == 3 else '' }}-pymysql"
+      - "python3-pymysql"
+
+- name: Ensure MySQL is started and enabled on boot.
+  service: "name={{ mysql_daemon }} state=started enabled={{ nextcloud_db_enabled_on_startup }}"
```
Comment (wiktor2200) [inline tasks/db_mysql.yml:18]: You should move this definition to handlers and use this notation:
```suggestion
  service: 
    name={{ mysql_daemon }}
    state=started
    enabled={{ nextcloud_db_enabled_on_startup }}"
```

File: molecule/default/converge.yml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+---
+- name: Converge
+  hosts: all
+  become: true
+
+  pre_tasks:
+    - name: Update apt cache.
+      ansible.builtin.apt: update_cache=yes cache_valid_time=600
```
Comment (wiktor2200) [inline molecule/default/converge.yml:8]:
```suggestion
      ansible.builtin.apt: 
        update_cache: true
        cache_valid_time: 600
```

**Review by wiktor2200 — CHANGES_REQUESTED** [review]
Please change to Ansible YAML syntax.

**Comment by aalaesar** [comment]
Hello @staticdev.
I see some issues  with the mysql authentication on the system.
Since the 6 years I've made this role I found that passing the password as environment  variable is secured enough : it doesn't not to appear in ps command and in ansible very verbose logs so I dumped the .my.cnf configuration on my others roles.
Maybe this would resolve the mess as the role knows the root password during the mysql installation.
How about I implement that ? This should simplify this part.

**Comment by staticdev** [comment]
> Hello @staticdev. I see some issues with the mysql authentication on the system. Since the 6 years I've made this role I found that passing the password as environment variable is secured enough : it does not to appear in ps command and in ansible very verbose logs so I dumped the .my.cnf configuration on my others roles. Maybe this would resolve the mess as the role knows the root password during the mysql installation. How about I implement that ? This should simplify this part.

Don't bother I am on it.. I refactored the password generation to be done on `setup_env`, this has some advantages and enables us to use roles that need password as a parameter.

**Comment by staticdev** [comment]
@aalaesar Mysql is also fixed.. the missing error is the one reported on https://github.com/aalaesar/install_nextcloud/issues/112
I can try to fix in a later PR.

File: README.md [hunk 4]
```diff
@@ -32,7 +32,7 @@ This role requires root access, so either configure it in your inventory files,
 - hosts: dnsserver
   become: true
   roles:
-    - role: aalaesar.install_nextcloud
+    - role: aalaesar.nextcloud
```
Comment (aalaesar) [inline README.md:35]: seems the name change was also commited in this branch.
can you revert it ? thanks :)

**Review by aalaesar — CHANGES_REQUESTED** [review]

File: README.md (hunk 4 above)
Comment (staticdev) [reply README.md]: @aalaesar this PR was synced automatically after the merge please check again.

**Pull request merged**